airflow/dag_processing/manager.py

```diff
@@ -1202,6 +1226,15 @@ def _kill_timed_out_processors(self):
         for proc in processors_to_remove:
             self._processors.pop(proc)
 
+    def _add_paths_to_queue(self, file_paths_to_enqueue: list[str], add_at_front: bool = False):
```

Comment:
I added this method since there were two places we added to the queue and both were virtually identical, except one added at the front, and one at the back. This also avoided repeating the update of the new metric tracking the queue size.

Reply:
NIT. Could we make `add_at_front` mandatory - i.e remove default and add it where it is called? This will add clarity to the places where it is used.

Reply:
Done